### ISSUE TITLE
Flat latent space

### ISSUE THREAD

**Issue by alexcwsmith** [LINCellularNeuroscience/VAME#23]
Hello,
I am wondering if you may be able to give insight to parameter tuning. Running with a top-down view produces results as expected, but I am getting weird results when I try to run on some DLC data that was acquired with a side view (with 9 points labeled, nose, tail-base, 3 points along the spine, and 4 paws). These videos are 30FPS, and if I train with time_window=15, I get a mostly flat reconstruction. I'm not positive how to interpret this, but I think this indicates that it is just not detecting any change in motion in the latent state? Would you recommend increasing the time window as a first step at improving this model?

In this experiment, about 1/2 the videos show the mice engaging in highly stereotyped behavior. I am wondering if that could contribute to a flat latent state, as there is not as much variation in animal motion in those videos. Would you recommend training a model on only non-experimental (i.e. non-stereotyped behavior) videos and then using that model to analyze the experimental videos? I'm not sure how I could quantify accuracy if I do not include the stereotyped behavior in the training of the model.

Thanks,
Alex

**Comment by KLuxem** [comment]
Hi Alex,

I apoligise for the late respond. I would definetely try to set the time window back to 30.  How many latent dimensions are you using to encode the behavior and how many datapoints do you have in your training dataset? Maybe also, try to set the `kmeans_lambda` to 0 in your config.yaml, just to see how the model behaves. 

For your second question, I would train the model on both datasets (non-experimental and experimental). 

Cheers,
Kevin

**Comment by alexcwsmith** [comment]
Great, thanks for the tips on things to try.

I have mostly left zdims=30. I have also tried 20 and 40, with no noticeable increase in performance. I'm not sure what a good value to try would be, so if you can recommend something there that would be awesome. One thing that I have found does increase performance is changing savgol_length=7 and savgol_order=3. I visualized a bunch of individual savgol filters on some of my datasets, and that looked good and performed well in the model, but honestly I've long since forgotten most of what I read on that.

In one dataset (recorded from a side view, camera 90 degree angle facing the mouse), I have about a million frames, (~800k in training set I believe), and have played with a variety of different numbers of features to consider on the labeled mouse. My mice have 9 labeled bodyparts, but I have played around with only training on as low as 4 of them (the 4 highest mean likelihood: nose, tail_base, spine1, spine2), and didn't get great results with that either. I've tested quite a few different combinations of body parts.

My second dataset is a top-down view with a more standard 6 labeled bodypartrs, and there I have about 700k total frames. This model seems to perform a bit better, but still have the problem of flat latency and not wowing performance.

Thanks for the tips, will try those and report back, and if you have any other suggestions please let me know!
